flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamPythonCalc.scala

```diff
@@ -0,0 +1,173 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.nodes.datastream
+
+import org.apache.calcite.plan.{RelOptCluster, RelTraitSet}
+import org.apache.calcite.rel.RelNode
+import org.apache.calcite.rel.core.Calc
+import org.apache.calcite.rex.{RexCall, RexInputRef, RexProgram}
+import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.streaming.api.datastream.DataStream
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator
+import org.apache.flink.table.api.StreamQueryConfig
+import org.apache.flink.table.calcite.FlinkTypeFactory
+import org.apache.flink.table.codegen.FunctionCodeGenerator
+import org.apache.flink.table.functions.python.PythonFunctionInfo
+import org.apache.flink.table.plan.nodes.CommonPythonCalc
+import org.apache.flink.table.plan.nodes.datastream.DataStreamPythonCalc._
+import org.apache.flink.table.plan.schema.RowSchema
+import org.apache.flink.table.planner.StreamPlanner
+import org.apache.flink.table.runtime.CRowProcessRunner
+import org.apache.flink.table.runtime.types.{CRow, CRowTypeInfo}
+import org.apache.flink.table.types.logical.RowType
+import org.apache.flink.table.types.utils.TypeConversions
+
+import scala.collection.JavaConversions._
+
+class DataStreamPythonCalc(
+    cluster: RelOptCluster,
+    traitSet: RelTraitSet,
+    input: RelNode,
+    inputSchema: RowSchema,
+    schema: RowSchema,
+    calcProgram: RexProgram,
+    ruleDescription: String)
+  extends DataStreamCalcBase(
+    cluster,
+    traitSet,
+    input,
+    inputSchema,
+    schema,
+    calcProgram,
+    ruleDescription)
+  with CommonPythonCalc {
+
+  override def copy(traitSet: RelTraitSet, child: RelNode, program: RexProgram): Calc = {
+    new DataStreamPythonCalc(
+      cluster,
+      traitSet,
+      child,
+      inputSchema,
+      schema,
+      program,
+      ruleDescription)
+  }
+
+  private lazy val pythonRexCalls = calcProgram.getProjectList
+    .map(calcProgram.expandLocalRef)
+    .filter(_.isInstanceOf[RexCall])
+    .map(_.asInstanceOf[RexCall])
+    .toArray
+
+  private lazy val (pythonUDFInputOffsets, pythonFunctionInfos) =
+    extractPythonScalarFunctionInfos(pythonRexCalls)
+
+  private lazy val resultProjectList = {
+    var idx = 0
+    calcProgram.getProjectList
+      .map(calcProgram.expandLocalRef)
+      .map {
+        case pythonCall: RexCall =>
+          val inputRef = new RexInputRef(input.getRowType.getFieldCount + idx, pythonCall.getType)
+          idx += 1
+          inputRef
+        case node => node
+      }
+  }
+
+  override def translateToPlan(
```

Comment:
Maybe we can remove the implementation of this method to the next PR. This method can not be tested yet. It's hard to make sure the correctness of these codes. What do you think?